google-cloud-storage/src/main/java/com/google/cloud/storage/GrpcStorageImpl.java

```diff
@@ -936,12 +937,59 @@ public boolean deleteDefaultAcl(String bucket, Entity entity) {
 
   @Override
   public Acl createDefaultAcl(String bucket, Acl acl) {
-    return throwNotYetImplemented(fmtMethodName("createDefaultAcl", String.class, Acl.class));
+    return updateDefaultAcl(bucket, acl);
   }
 
   @Override
   public Acl updateDefaultAcl(String bucket, Acl acl) {
-    return throwNotYetImplemented(fmtMethodName("updateDefaultAcl", String.class, Acl.class));
+    try {
+      com.google.storage.v2.Bucket resp = getBucketDefaultAcls(bucket);
+      ObjectAccessControl encode = codecs.objectAcl().encode(acl);
+      String entity = encode.getEntity();
+
+      Predicate<ObjectAccessControl> entityPredicate = objectAclEntityOrAltEq(entity);
+
+      ImmutableList<ObjectAccessControl> collect =
+          Streams.concat(
+                  resp.getDefaultObjectAclList().stream().filter(entityPredicate.negate()),
+                  Stream.of(encode))
+              .collect(ImmutableList.toImmutableList());
+
+      com.google.storage.v2.Bucket update =
+          com.google.storage.v2.Bucket.newBuilder()
+              .setName(bucketNameCodec.encode(bucket))
+              .addAllDefaultObjectAcl(collect)
+              .build();
+      Opts<BucketTargetOpt> opts =
+          Opts.from(
+              UnifiedOpts.fields(ImmutableSet.of(BucketField.DEFAULT_OBJECT_ACL)),
+              UnifiedOpts.metagenerationMatch(resp.getMetageneration()));
+      UpdateBucketRequest req =
+          opts.updateBucketsRequest()
+              .apply(UpdateBucketRequest.newBuilder())
+              .setBucket(update)
+              .build();
+
+      GrpcCallContext grpcCallContext = GrpcCallContext.createDefault();
+      com.google.storage.v2.Bucket updateResult =
+          Retrying.run(
+              getOptions(),
+              retryAlgorithmManager.getFor(req),
+              () -> storageClient.updateBucketCallable().call(req, grpcCallContext),
+              Decoder.identity());
+
+      //noinspection DataFlowIssue
+      Optional<Acl> first =
+          updateResult.getDefaultObjectAclList().stream()
+              .filter(entityPredicate)
+              .findFirst()
+              .map(codecs.objectAcl()::decode);
+
+      return first.orElseThrow(
+          () -> new StorageException(404, "Acl update call success, but not in response"));
```

Comment:
When would this occur?

Reply:
Hopefully never, but since we have to search for the updated acl in the response, if the response somehow doesn't included the updated acl we need to signal that somehow. 404 here doesn't feel totally correct, but I'm not sure 500 is better. What do you think?

Reply:
Not sure it's necessary tbh, you can only have up to 100 acls associated to a bucket or object; 
![image](https://user-images.githubusercontent.com/1907648/207456309-4bf0ccea-f0af-4422-8683-0025cb0dcd7b.png)

Thinking we just rely on the service response cod in this case unless I'm misunderstanding.

Reply:
I spoke with @frankyn offline, we came to the consensus that instead of returning an HTTP status code here, insead we will return undefined `0` to signal that it's purely a client side issue, not any system issue.